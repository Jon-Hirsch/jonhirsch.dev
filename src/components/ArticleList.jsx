import { Link } from "gatsby";
import React from "react";
import PropTypes from "prop-types";

export default function ArticleList(props) {
  return (
    <div className="articles-list">
      {props.nodes.map((node) => (
        <div key={node.id} className="article-div" data-testid="article-div">
          <h3 className="article-link">
            <Link to={`/${node.slug}`}>{node.frontmatter.title}</Link>
          </h3>
          <p className="article-description">{node.frontmatter.description}</p>
        </div>
      ))}
    </div>
  );
}

ArticleList.propTypes = {
  nodes: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      slug: PropTypes.string.isRequired,
      frontmatter: PropTypes.shape({
        title: PropTypes.string.isRequired,
        description: PropTypes.string.isRequired,
      }).isRequired,
    })
  ).isRequired,
};
