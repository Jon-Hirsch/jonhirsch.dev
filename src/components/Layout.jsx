import { Link } from "gatsby";
import React, { useEffect, useState } from "react";
import PropTypes from "prop-types";
import "@fontsource/quattrocento-sans";
import "@fontsource/quicksand";
import "@fontsource/roboto";
import "../styles/styles.scss";
import Helmet from "react-helmet";
import { CSSTransition } from "react-transition-group";

export default function Layout(props) {
  const [showMenu, setShowMenu] = useState(false);
  const [isScrolled, setIsScrolled] = useState(false);

  useEffect(() => {
    function handleScroll() {
      if (!isScrolled && window.scrollY > 35) {
        setIsScrolled(true);
      } else if (isScrolled && window.scrollY <= 35) {
        setIsScrolled(false);
      }
    }

    window.addEventListener("scroll", handleScroll);
    return () => {
      window.removeEventListener("scroll", handleScroll);
    };
  }, [isScrolled]);

  return (
    <>
      <div className="page">
        <Helmet>
          <title>JonHirsch.dev</title>
        </Helmet>
        <div
          className={isScrolled ? "page-header scrolled" : "page-header"}
          data-testid="page-header"
        >
          <Link className="home-link" to="/">
            jonhirsch.dev
          </Link>
          <div className="navigation">
            <Link to="/">Home</Link>
            <Link to="/personal-projects">Personal Projects</Link>
            <Link to="/articles">Articles</Link>
            <Link to="/search">Search</Link>
          </div>
          <button
            className="nav-hamburger"
            aria-label="open navigation menu"
            data-testid="nav-button"
            onClick={() => setShowMenu(true)}
          >
            <div className="hamburger-line"></div>
            <div className="hamburger-line"></div>
            <div className="hamburger-line"></div>
          </button>
        </div>
        <div className={isScrolled ? "page-content scrolled" : "page-content"}>
          {props.children}
        </div>
      </div>
      <CSSTransition
        in={showMenu}
        classNames="menu-transition"
        timeout={400}
        mountOnEnter={true}
        unmountOnExit={true}
      >
        {/* disable a11y warnings on this line. The close button is included to handle a11y navigation */}
        {/* eslint-disable-next-line */}
        <div className="menu-overlay" onClick={() => setShowMenu(false)}>
          <div className="close-menu-button-container">
            <button
              className="close-menu-button"
              data-testid="close-menu-button"
              aria-label="close navigation menu"
              onClick={() => setShowMenu(false)}
            >
              <div className="close-button-line"></div>
              <div className="close-button-line"></div>
            </button>
          </div>
          <div className="menu-links" data-testid="menu-links">
            <Link to="/">Home</Link>
            <Link to="/personal-projects">Personal Projects</Link>
            <Link to="/articles">Articles</Link>
            <Link to="/search">Search</Link>
          </div>
        </div>
      </CSSTransition>
    </>
  );
}

Layout.propTypes = {
  children: PropTypes.array,
};
