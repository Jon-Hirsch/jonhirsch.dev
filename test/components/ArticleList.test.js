import React from "react";
import { render, cleanup, fireEvent, waitFor } from "@testing-library/react";
import Layout from "../../src/components/Layout";
import ArticleList from "../../src/components/ArticleList";

afterEach(cleanup);

describe("ArticleList", () => {
  test("should render a list of articles", () => {
    const nodes = [
      {
        id: 1,
        slug: "/1",
        frontmatter: {
          title: "title1",
          description: "description 1",
        },
      },
      {
        id: 2,
        slug: "/2",
        frontmatter: {
          title: "title2",
          description: "description 2",
        },
      },
      {
        id: 3,
        slug: "/3",
        frontmatter: {
          title: "title3",
          description: "description 3",
        },
      },
    ];
    const { container } = render(<ArticleList nodes={nodes} />);

    expect(container).toMatchSnapshot();
  });
});
