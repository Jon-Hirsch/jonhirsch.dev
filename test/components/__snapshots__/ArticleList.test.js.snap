// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList should render a list of articles 1`] = `
<div>
  <div
    class="articles-list"
  >
    <div
      class="article-div"
    >
      <h3
        class="article-link"
      >
        <a
          href="//1"
        >
          title1
        </a>
      </h3>
      <p
        class="article-description"
      >
        description 1
      </p>
    </div>
    <div
      class="article-div"
    >
      <h3
        class="article-link"
      >
        <a
          href="//2"
        >
          title2
        </a>
      </h3>
      <p
        class="article-description"
      >
        description 2
      </p>
    </div>
    <div
      class="article-div"
    >
      <h3
        class="article-link"
      >
        <a
          href="//3"
        >
          title3
        </a>
      </h3>
      <p
        class="article-description"
      >
        description 3
      </p>
    </div>
  </div>
</div>
`;
